cinema/views.py

```diff
@@ -14,45 +23,64 @@
     MovieSessionSerializer,
     MovieSessionListSerializer,
     MovieDetailSerializer,
-    MovieSessionDetailSerializer,
     MovieListSerializer,
     OrderSerializer,
-    OrderListSerializer,
+    OrderListSerializer, MovieSessionDetailSerializer,
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
instead of duplicating code, you can set DEFAULT_PERMISSION_CLASSES and DEFAULT_AUTHENTICATION_CLASSES in REST_FRAMEWORK constant in settings.py and override these attributes where needed